src/Data/Foldable.purs

```diff
@@ -169,6 +170,10 @@ instance foldableMultiplicative :: Foldable Multiplicative where
 fold :: forall f m. Foldable f => Monoid m => f m -> m
 fold = foldMap id
 
+-- | Similar to 'foldl', but the result is encapsulated in a monad. 
```

Comment:
We should add a note that this isn't stack-safe in general, for any monad which involves building up a thunk, like `Eff`.